system/test_combiners.py

```diff
@@ -3,55 +3,80 @@
-  assert body.count("looking for media &#39;print&#39; but found media=&#39;&#39;.") > 0
-  assert body.count("Could not combine over barrier: noscript") > 0
-  assert body.count("Could not combine over barrier: inline style") > 0
-  assert body.count("Could not combine over barrier: IE directive") > 0
 
+def test_combine_css_combines_4_CSS_files_into_1():
+    url = "%s/combine_css.html?PageSpeedFilters=combine_css" % test_fixtures.EXAMPLE_ROOT
+    helpers.get_until_primary(
+        url,
+        {},
+        lambda response,
+        body: body.count("text/css") == 1)
+
+
+def test_combine_css_without_hash_field_should_404():
+    url = "%s/styles/yellow.css+blue.css.pagespeed.cc..css" % test_fixtures.REWRITTEN_ROOT
+    resp, body = helpers.get_primary(url, {})
+    assert resp.status == 404
+
+
+def test_fetch_large_css_combine_url():
+    url = "%s/styles/yellow.css+blue.css+big.css+" \
+        "bold.css+yellow.css+blue.css+big.css+bold.css+yellow.css+blue.css+" \
+        "big.css+bold.css+yellow.css+blue.css+big.css+bold.css+yellow.css+blue.css+" \
+        "big.css+bold.css+yellow.css+blue.css+big.css+bold.css+yellow.css+blue.css+" \
+        "big.css+bold.css+yellow.css+blue.css+big.css+bold.css+yellow.css+blue.css+" \
+        "big.css+bold.css+yellow.css+blue.css+big.css+bold.css+yellow.css+blue.css+" \
+        "big.css+bold.css+yellow.css+blue.css+big.css+bold.css+yellow.css+blue.css+" \
+        "big.css+bold.css+yellow.css+blue.css+big.css+bold.css+yellow.css+blue.css+" \
+        "big.css+bold.css+yellow.css+blue.css+big.css+bold.css+yellow.css+blue.css+" \
+        "big.css+bold.css+yellow.css+blue.css+big.css+" \
+        "bold.css.pagespeed.cc.46IlzLf_NK.css" % test_fixtures.REWRITTEN_ROOT
+
+    resp, body = helpers.get_primary(url, {})
+    assert resp.status == 200
+    assert len(body.splitlines()) > 900
+
+
+def test_combine_javascript_combines_2_JS_files_into_1():
+    url = "%s/combine_javascript.html?PageSpeedFilters=combine_javascript" % test_fixtures.EXAMPLE_ROOT
+    helpers.get_until_primary(
+        url,
+        {},
+        lambda response,
+        body: body.count("src=") == 1)
+
+
+def test_combine_javascript_with_long_URL_still_works():
+    url = "%s/combine_js_very_many.html?PageSpeedFilters=combine_javascript" % test_fixtures.TEST_ROOT
+    helpers.get_until_primary(
+        url,
+        {},
+        lambda response,
+        body: body.count("src=") == 4)
+
+
+def test_combine_heads_combines_2_heads_into_1():
+    url = "%s/combine_heads.html?PageSpeedFilters=combine_heads" % test_fixtures.EXAMPLE_ROOT
+    resp, body = helpers.get_primary(url)
+    assert body.count("<head>") == 1
+
+
+def test_combine_css_debug_filter():
+    url = "%s/combine_css_debug.html?PageSpeedFilters=combine_css,debug" % test_fixtures.EXAMPLE_ROOT
+    helpers.get_until_primary(
+        url, {},
+        lambda response, body: body.count(
+            "styles/yellow.css+blue.css+big.css+bold.css.pagespeed.cc") == 1)
```

Comment:
@jeffkaufman 
re: This would be good to have a helper function for. Something like:
helpers.get_until_count(url, string_to_match, count)

What do you think about adding an optional 'checker' to the arguments?:
```python
    helpers.get_primary(
        "%s/add_instrumentation.html?PageSpeedFilters=" %
        config.EXAMPLE_ROOT, check = helpers.CheckSubstringCountEquals("<script", 1))
```

CheckSubstringCountEquals could be defined like this in helpers.py, and we could add similar methods for checks involving regular expressions and so on?:

```python
def CheckSubstringCountEquals(str, count):
    return lambda resp, body: body.count(str) == count
```